coriolis/providers/base.py

```diff
@@ -309,39 +309,47 @@ def restore_replica_disk_snapshots(self, ctxt, connection_info,
 class BaseExportProvider(BaseInstanceProvider):
 
     @abc.abstractmethod
-    def export_instance(self, ctxt, connection_info, instance_name,
-                        export_path):
+    def export_instance(self, ctxt, connection_info, source_environment,
+                        instance_name, export_path):
         """Exports the given instance.
 
          Exports the instance given by its name from the given source cloud
         to the provided export directory path using the given connection info.
         """
         pass
 
+    @abc.abstractmethod
+    def get_source_environment_schema(self):
```

Comment:
Please define a separate `BaseExportInstanceProvider` class for this method and make both `BaseExportProvider` and `BaseReplicaExportProvider` inherit it (this way it's only required in the `BaseExportProvider` ABC and not the Replica one as well)